ext/opcache/Optimizer/pass1_5.c

```diff
@@ -309,6 +307,11 @@ void zend_optimizer_pass1(zend_op_array *op_array, zend_optimizer_ctx *ctx)
 						}
 					}
 				} else if (op_array->scope &&
+					ZEND_OP1_TYPE(opline) == IS_UNUSED &&
+					(opline->op1.num & ~ZEND_FETCH_CLASS_NO_AUTOLOAD) == ZEND_FETCH_CLASS_SELF) {
```

Comment:
thanks. fixed now.

On Wed, Oct 28, 2015 at 12:29 PM, Nikita Popov <notifications@github.com>
wrote:

> In ext/opcache/Optimizer/pass1_5.c
> <https://github.com/php/php-src/pull/1604#discussion_r43232477>:
>
> > @@ -309,6 +307,11 @@ void zend_optimizer_pass1(zend_op_array *op_array, zend_optimizer_ctx *ctx)
> >  						}
> >  					}
> >  				} else if (op_array->scope &&
> > +					ZEND_OP1_TYPE(opline) == IS_UNUSED &&
> > +					(opline->op1.num & ~ZEND_FETCH_CLASS_NO_AUTOLOAD) == ZEND_FETCH_CLASS_SELF) {
>
> I mean that I suspect (but haven't verified) that this optimization (both
> in this branch and previously) currently doesn't trigger, because we check
> for (num & ~ZEND_FETCH_CLASS_NO_AUTOLOAD) == ZEND_FETCH_CLASS_SELF, while
> num will also contain a ZEND_FETCH_CLASS_EXCEPTION flag, so the check
> will never be true.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/php/php-src/pull/1604/files#r43232477>.
>